src/auth/authclient.ts

```diff
@@ -147,4 +147,30 @@ export abstract class AuthClient
     }
     return headers;
   }
+
+  /**
+   * Gets the quota project id on the credential
+   */
+  getQuotaProjectId() {
+    return this.quotaProjectId;
+  }
+
+  /**
+   * Sets the quota project id on the credential
+   * @param quotaProjectId The quota project id to set
+   */
+  setQuotaProjectId(quotaProjectId?: string): void {
+    this.quotaProjectId = quotaProjectId;
+  }
+
+  /**
+   * Sets the quota project id from the GOOGLE_CLOUD_QUOTA_PROJECT
+   * environment variable on the credential
+   */
+  setQuotaProjectIdFromEnvironment(): void {
+    const quotaProjectIdFromEnv = process.env['GOOGLE_CLOUD_QUOTA_PROJECT'];
+    if (quotaProjectIdFromEnv) {
+      this.setQuotaProjectId(quotaProjectIdFromEnv);
+    }
+  }
```

Comment:
Going along with the other two comments (removing the get/set and resetting `quotaProjectId` to the env on every call), perhaps this method should be removed and the `env` variable be the default for `quotaProjectId`:
```
quotaProjectId = process.env['GOOGLE_CLOUD_QUOTA_PROJECT'] || '';
```

Reply:
If we do this way instead of setting from ADC method, then in some creds, this value could be overriden by a value from the json file.

Reply:
Since our preference is for the environment variable over the JSON file when they're both available, then we should check to see if it exists before using the JSON input. Fortunately, there are just a few references to make this check (as seen on the left side bar):

<img width="1144" alt="Screen Shot 2022-10-26 at 5 58 05 PM" src="https://user-images.githubusercontent.com/10792787/198166288-f93c5096-e6c7-4029-8707-afeae2598a3a.png">

Reply:
Currently we are using environment variable only when the credential is returned through ADC. The priority order might be different when the user is explicitly creating the credential (including loading from a json file). So it will be good to restrict this change within scope of ADC. For more details, checkout [go/quota-env](https://docs.google.com/document/d/1_ZwNyfzhp4AkYm3KCfuPpAtACejVQgaGG9mY7daXmYU/edit?resourcekey=0-uhZZSFONLMSXvAQtXF2BnQ#bookmark=id.tu1rns2ty0fa)